app/src/main/java/org/simple/clinic/instantsearch/InstantSearchEffectHandler.kt

```diff
@@ -36,6 +36,7 @@ class InstantSearchEffectHandler @AssistedInject constructor(
       .addConsumer(ShowNoPatientsInFacility::class.java, { uiActions.showNoPatientsInFacility(it.facility) }, schedulers.ui())
       .addAction(ShowNoSearchResults::class.java, { uiActions.showNoSearchResults() }, schedulers.ui())
       .addAction(HideNoPatientsInFacility::class.java, uiActions::hideNoPatientsInFacility, schedulers.ui())
+      .addAction(HideNoSearchResults::class.java, uiActions::hideNoSearchResults, schedulers.ui())
```

Comment:
These should be handled in the UI renderer and not via effects. UI actions are meant for things that are not part of the screen as such, like showing a modal or navigating to a different screen.

Reply:
Hopefully, this [comment ](https://github.com/simpledotorg/simple-android/pull/2257#discussion_r551903893) clarifies why I went with this usage.